File: src/main/java/rest/controller/MockedPersonController.java
package rest.controller;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
import rest.model.Person;

import java.util.List;

@RestController
@RequestMapping("/mocked/person")
public class MockedPersonController {

    private List<Person> persons = List.of(
            new Person(1L, "Alexander", "Svensson"),
            new Person(2L, "Anna", "Andersson"),
            new Person(3L, "Mikael", "Smith"),
            new Person(4L, "Mia", "Pia")
    );

    @RequestMapping(method = RequestMethod.GET)
    public List<Person> getPersons() {
        return persons;
    }

    @RequestMapping(method = RequestMethod.GET, value = "/ping")
    public String ping() {
        return "Hello world";
    }
}

File: src/main/java/rest/controller/MockedPersonController.kt
package rest.controller

import org.springframework.web.bind.annotation.*
import rest.model.Person
import java.util.*

@RestController
@RequestMapping("/mocked/person")
class MockedPersonController {
    @get:RequestMapping(method = [RequestMethod.GET])
    val persons = listOf<Person>(
            Person(1L, "Alexander", "Svensson"),
            Person(2L, "Anna", "Andersson"),
            Person(3L, "Mikael", "Smith"),
            Person(4L, "Mia", "Pia")
    )

    @RequestMapping(method = [RequestMethod.GET], value = ["/{id}"])
    fun getPerson(@PathVariable id: Long): Optional<Person> {
        return persons
                .stream()
                .filter { person: Person -> person.id == id }
                .findFirst()
    }

    @RequestMapping(method = [RequestMethod.POST])
    fun createPerson(@RequestBody person: Person): Person {
        return Person(2L, person.firstname, person.lastname)
    }

    @RequestMapping(method = [RequestMethod.PUT], value = ["/{id}"])
    fun updatePerson(@PathVariable id: Long?, @RequestBody person: Person): Person {
        return Person(id, person.firstname, person.lastname)
    }

    @RequestMapping(method = [RequestMethod.DELETE], value = ["/{id}"])
    fun deletePerson(@PathVariable id: Long?) {
        //        should call service to delete entity
    }

    @RequestMapping(method = [RequestMethod.GET], value = ["/ping"])
    fun ping(): String {
        return "Hello world"
    }
}